knowledge-base/textarea-indicate-when-maxlength-is-reached.md

```diff
@@ -0,0 +1,135 @@
+
+To notify the user that they have reached the `MaxLength` value:
+
+* Declare the desired message in an `if` block. 
+* Display this message if the length of the TextArea value equals the specified `MaxLength` value. 
+* Add the desired CSS rules or classes to style the message as needed.
+
+To let the user know how many characters they have left to type, render the TextArea value length and the `MaxLength` value - this will serve as a character counter.
+
+>caption Show a character counter and a message when max is reached
+
+````CSHTML
+@using System.ComponentModel.DataAnnotations
+
+@if (string.IsNullOrEmpty(FormSubmitMessage))
+{
+    <TelerikForm Model="@FormModel"
+                 OnValidSubmit="@OnValidSubmit"
+                 OnInvalidSubmit="@OnInvalidSubmit"
+                 Width="400px">
+        <FormValidation>
+            <DataAnnotationsValidator />
+        </FormValidation>
+        <FormItems>
+            <FormItem>
+                <Template>
+                    <label for="sendInvitation">Send Invitation:</label>
+                    <TelerikTextArea Id="sendInvitation"
+                                     Value="@FormModel.Text"
+                                     ValueExpression="@(() => FormModel.Text)"
+                                     ValueChanged="@OnValueChanged"
+                                     Placeholder="Enter your text here"
+                                     MaxLength="@MaxLength">
+                    </TelerikTextArea>
+                    <div class="textarea-info">
+                        <TelerikValidationMessage For="@(() => FormModel.Text)" />
+                        @if (TextLength == MaxLength)
+                        {
+                            <div class="k-form-hint k-form-error">You reached the maximum allowed characters.</div>
+                        }
+                        <span class="character-counter k-form-hint @(TextLength == MaxLength ? "k-form-error" : "")">@($"{TextLength} / {MaxLength}")</span>
+                    </div>
+                </Template>
+            </FormItem>
+        </FormItems>
+        <FormButtons>
+            <TelerikButton ButtonType="@ButtonType.Submit" ThemeColor="primary">Send</TelerikButton>
+        </FormButtons>
+    </TelerikForm>
+}
+else
+{
+    @FormSubmitMessage
+}
+
+<style>
+    .textarea-info {
+        display: flex;
+        justify-content: space-between;
+    }
+
+    .character-counter {
+        margin-left: auto;
+    }
+</style>
+
+@code {
+    public class ValidationModel
+    {
+        [Required(ErrorMessage = "Please enter a text.")]
+        public string Text { get; set; }
+    }
+
+    public ValidationModel FormModel = new ValidationModel();
+
+    public string FormSubmitMessage { get; set; }
+
+    public int TextLength => FormModel?.Text?.Length ?? 0;
+
+    public int MaxLength { get; set; } = 100;
```

Comment:
Use a smaller `MaxLength` value, so that developers can test this more easily. Currently I have to type something, then copy-paste it multiple times.